CenterForOpenScience/ember-osf-web
# [ENG-4153] File detail view metadata

-   Ticket: [ENG-4153]

## Purpose

Add the ability to view metadata on a file detail page. Tests and styling to come in future PRs.

## Summary of Changes

1. Fix up file metadata manager
2. Add metadata section to file detail page
3. Fill out metadata section with file and target metadata
4. Fix file's guid creation to properly add a guid attribute to the file

## Screenshot(s)

Project file:
<img width="717" alt="Screen Shot 2022-11-15 at 10 44 29 AM" src="https://user-images.githubusercontent.com/6599111/201971901-bb80f604-9742-424a-bf8a-a1615d616bf3.png">
<img width="2031" alt="Screen Shot 2022-11-15 at 10 44 17 AM" src="https://user-images.githubusercontent.com/6599111/201971905-84145927-dc18-4d82-8d5d-649b0d14d8bd.png">

Registration file:
<img width="1914" alt="Screen Shot 2022-11-15 at 11 17 27 AM" src="https://user-images.githubusercontent.com/6599111/201971971-dec1ba32-1818-4112-afec-507c50b92e14.png">


## Side Effects

This shouldn't have any side effects.

## QA Notes

* The file metadata should match what's saved in the file, and the project/component/registration metadata should match what's in the files target node.
* The metadata tab should be out by default on a desktop screen. Shrinking the screen to a mobile size will keep the metadata tab as the active thing to view, but if you open the page in mobile size, the file rendering will be the thing to show.

[ENG-4153]: https://openscience.atlassian.net/browse/ENG-4153?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
